docs/proposals/version_documentation.md

```diff
@@ -0,0 +1,91 @@
+
+### Documentation Structure
+
+During the course of my research, I was able to deduce three different approaches of which we can solve this documentation structure issue.
+
+1. Sub branching: Creating different sub branches in the content folder of the Thanos repository. Since the current architecture of Hugo is this:
+
+|- archetypes
+|- config.toml
+|- content
+|- data
+|- layouts
+|- static
+|- themes
+
+You can add aditional versioned content in different folders within the content directory. For example:
+
+content
+|- docs
+|- master
+|- docs-v2
+|- docs-v3
+|- other-folder-for-other-releases
+
+_Note: The current version lives in the `docs` folder_
+
+#### Example:
+
+For `master` version we would be able to use the above links via the following path
+
+```
+/docs/master/components/
+```
+
+And for a minor version like `1.x`:
+
+```
+/docs/docs-v2/components/
+```
+
+2. Version Branching: You can decide to make each individual version to be a separate branch of it's own in the master repository.
+
+3. We can as well use templates/themes that have these processes automated e.g docsy, docusaurus, etc
+
+### Workflow
+
+Choosing a workflow solely depends on the kind of documentation structure we implement in our repositoory. The Sub branching documentation structure, which was listed above, can utilize the central workflow of which everything is tied to the master repository. You can make changes by pulling from the master, edit, and push new changes back to master. We can as well decide to implement the Version Branching workflow. In this workflow, You make each individual version to be a separate branch of it's own in the master repository. So each time a particular version of docs is selected from the version picker tool, it loads that particular branch.
+
+We can as well implement the Automated Scripting workflow. This allows us to use an automated script to get each content from each individual branch, and merge it to the latest branch.
+
+### Building a versioning plugin/extension
+
+Creating a plugin that can automate these process would save us alot of development time and stress. This approach promises to be very useful when it comes to versioning different release in the Thanos website. This can be achieved by implementing any of the following possible approach:
+
+1. Building a versioning plugin that can be integrated in different text editors which will help automate the entire versioning process.
+2. Building a script that will automate these processes.
+3. Building an interface that can automate these processes by just one click of a button.
+
+The workflow for versioning using the plugin/script will look like this:
+
+1. Create the latest version(master) of the docs by cloning content/docs of the immediate previous release into the latest version(master).
```

Comment:
So the latest version of docs would point to latest release, i.e. thanos v0.12.2? or would it point to the master branch in git?

Reply:
The latest version of docs would point to latest release. From my understanding of the Thanos git workflow, each release has it's own branch. Inotherwords, each version would represent individual release branch, i.e. Release 1.x. 

@povilasv Your thoughts would be really helpful.

Reply:
Yup that's correct, I think then maybe we need to change wording of that sentence a bit. Maybe something like this would do:
```
Create the latest version of the docs by cloning content of the master branch.
```

Reply:
Alright. Would effect the changes as soon as possible. Thanks for the feedback

Reply:
:+1: no rush, thanks for doing this!

Reply:
It's such an honour for me to contribute to this amazing project!